crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1070 @@
+    /// ````
+    ///
+    /// [turbo]: https://ai.googleblog.com/2019/08/turbo-improved-rainbow-colormap-for.html
+    #[func]
+    fn turbo(
+        #[default(Spanned::new(20, Span::detached()))] stops: Spanned<i64>,
+    ) -> SourceResult<Value> {
+        fn at(t: f32) -> Rgba {
+            let t = t.clamp(0.0, 1.0);
+            let r = (34.61
+                + t * (1172.33
+                    - t * (10793.56 - t * (33300.12 - t * (38394.49 - t * 14825.05)))))
+                .round();
+            let g = (23.31
+                + t * (557.33
+                    + t * (1225.33 - t * (3574.96 - t * (1073.77 + t * 707.56)))))
+                .round();
+            let b = (27.2
+                + t * (3211.1
+                    - t * (15327.97 - t * (27814.0 - t * (22569.18 - t * 6838.66)))))
+                .round();
+
+            Rgba::new(r / 255.0, g / 255.0, b / 255.0, 1.0)
+        }
+
+        if stops.v < 2 {
+            bail!(stops.span, "number of stops must be bigger or equal to 2");
+        }
+
+        Ok(Array::from(
+            (0..stops.v)
+                .map(|i| {
+                    let t = i as f64 / (stops.v - 1) as f64;
+                    Stop::new(Color::Rgba(at(t as f32)), t).into_value()
+                })
+                .collect::<EcoVec<_>>(),
+        )
+        .into_value())
+    }
+
+    /// Creates a [cividis] stop list.
+    ///
+    /// You can control the number of stops in the gradient using the `stops` parameter, by default it is set to 20.
+    ///
+    /// ```example
+    /// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.cividis(10)))
+    /// ````
+    ///
+    /// [cividis]: https://bids.github.io/colormap/
+    #[func]
+    fn cividis(
+        #[default(Spanned::new(20, Span::detached()))] stops: Spanned<i64>,
+    ) -> SourceResult<Value> {
+        fn at(t: f32) -> Rgba {
+            let t = t.clamp(0.0, 1.0);
+            let r = -4.54
+                - t * (35.34
+                    - t * (2381.73 - t * (6402.7 - t * (7024.72 - t * 2710.57))));
+            let g = 32.49
+                + t * (170.73 + t * (52.82 - t * (131.46 - t * (176.58 - t * 67.37))));
+            let b = 81.24
+                + t * (442.36
+                    - t * (2482.43 - t * (6167.24 - t * (6614.94 - t * 2475.67))));
+
+            Rgba::new(r / 255.0, g / 255.0, b / 255.0, 1.0)
+        }
+
+        if stops.v < 2 {
+            bail!(stops.span, "number of stops must be bigger or equal to 2");
+        }
+
+        Ok(Array::from(
+            (0..stops.v)
+                .map(|i| {
+                    let t = i as f64 / (stops.v - 1) as f64;
+                    Stop::new(Color::Rgba(at(t as f32)), t).into_value()
+                })
+                .collect::<EcoVec<_>>(),
+        )
+        .into_value())
+    }
+
+    /// Creates a rainbow list of color stops with the given parameters.
```

Comment:
Perhaps this is clearer wording?
```suggestion
    /// Creates a list of rainbow color stops with the given parameters.
```